dockstore-common/src/main/java/io/dockstore/common/yaml/constraints/BaseConstraintValidator.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2023 OICR, UCSC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.dockstore.common.yaml.constraints;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+/**
+ * Base class for most ConstraintValidators, with an empty initializer
+ * code that handles null values correctly, and a helper method
+ * addConstraintViolation() that creates a constraint violation with
+ * the specified message.
+ */
+public abstract class BaseConstraintValidator<AnnotationT extends java.lang.annotation.Annotation, TargetT> implements ConstraintValidator<AnnotationT, TargetT> {
+    @Override
+    public void initialize(AnnotationT constraintAnnotation) {
+        // Intentionally empty
+    }
+
+    @Override
+    public boolean isValid(TargetT target, ConstraintValidatorContext context) {
+        if (target == null) {
```

Comment:
Noticed this one after my comment about null shortName. Still wondering why null is valid though... It seems intentional give your class JavaDoc comment, but I don't understand why.

Reply:
This is intentional: the general pattern is for validators to consider `null` values as valid, so that you can either explicitly disallow them with an accompanying `NotNull` annotation, or do nothing else and let them through.